### PR TITLE
remember product price in DB mongo [mt-2]

### LINKED ISSUE #11: remember product price in DB (mongo)
remember product price in DB (mongo) mt-2

### PR DESCRIPTION
**Pull request by yuriyFsd**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 /node_modules
-.env
\ No newline at end of file
+.env
+.env.example
```
Comment (lex-rio) [inline .gitignore]: not a good idea

File: index.js [hunk 2]
```diff
@@ -13,65 +14,66 @@ app.use((err, req, res, next) => {
...
+      telegram.log(goodInfo)
+      goodsStoreService.updateProductsInStorage(url, {prices: actualInfo[index].prices})
+      goodsStoreService.updateProductsInStorage(url, { status: actualInfo[index].status })
+    } else if (status !== actualInfo[index].status) { //if no discount but status updated
+      telegram.log(goodInfo)
+      goodsStoreService.updateProductsInStorage(url, {
+        status: actualInfo[index].status
+      })
+    }
+    index++
```
Comment (lex-rio) [inline index.js]: map has index as [2nd argument](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map#index

File: index.js [hunk 3]
```diff
@@ -13,65 +14,66 @@ app.use((err, req, res, next) => {
...
+ * @return {Promise<Array<string>>} An array of strings containing the updated product information.
+ */
+async function monitor() {
+  const goodsInitial = await getProducts()
+  const actualInfo = await getActualGoodsInfo(goodsInitial)
+  let index = 0
+
+  const updatedGoods = goodsInitial.map(async ({ url, title, prices, status }) => {
+    const goodInfo = `${url}  : ${actualInfo[index].status} ${actualInfo[index]?.prices || ''}`
+    if (actualInfo[index]?.prices &&
```
Comment (lex-rio) [inline index.js:55]: I would put each branch of this if statement to separate function for better readability

File: services/goods.extraction.service.js [hunk 4]
```diff
@@ -1,85 +1,95 @@
...
     }
+    const productHtmlBlock = htmlSource.substring(productOpenDivIndex, productDivCloseIndex)
+    const productCost = this.extractValues(productHtmlBlock)
 
-    /**
-     * @param {string} htmlSource
-     */
-    isProductNonAvailable(htmlSource) {
-        return htmlSource.includes(NOTAVAILABLE_TEXT)
+    if (productCost[0] > 1000000) {
```
Comment (lex-rio) [inline services/goods.extraction.service.js]: magic number

File: services/goods.extraction.service.js [hunk 5]
```diff
@@ -1,85 +1,95 @@
...
-                        return i
-                    }
-                }
-            }
+  /**
+   * @param {string} htmlString
+   * @param {any} openDivIndex
+   * @param {string} [tagName='div']
+   */
+  findClosingTagIndex(htmlString, openDivIndex, tagName = 'div') {
```
Comment (lex-rio) [inline services/goods.extraction.service.js:56]: it's hard to read this function. I would consider refactoring

File: services/http.client.service.js [hunk 6]
```diff
@@ -2,7 +2,10 @@ import fetch from 'node-fetch' //import fetch from 'node-fetch'
 
 export class HttpClientService {
     async getProductPageHtml(url) {
-        const response = await fetch(url)
+        let response = await fetch(url)
+        if (response.status !== 200) {
```
Comment (lex-rio) [inline services/http.client.service.js:6]: comment would be useful here

Comment (yuriyFsd) [reply services/http.client.service.js:6]: fixed

File: .gitignore (hunk 1 above)
Comment (yuriyFsd) [reply .gitignore]: fixed

File: index.js (hunk 2 above)
Comment (yuriyFsd) [reply index.js]: fixed

File: services/goods.extraction.service.js (hunk 4 above)
Comment (yuriyFsd) [reply services/goods.extraction.service.js]: fixed

File: index.js (hunk 3 above)
Comment (yuriyFsd) [reply index.js:55]: fixed

File: services/goods.extraction.service.js (hunk 5 above)
Comment (yuriyFsd) [reply services/goods.extraction.service.js:56]: at this moment I don't see a way how to refactor it, maybe next time I do

**Pull request merged**